modules/react/popup/stories/Popup.stories.mdx

```diff
@@ -0,0 +1,318 @@
+container component isn't needed.
+
+<ArgsTable of={Popup} />
+
+## Popup.Target
+
+A `Popup.Target` is any element that is meant to show the Popup. The default component rendered by
+this component is a `Button` element. You can override this by passing the desired component via
+`as`. Many examples above use `as={DeleteButton}`. If you want to render an `IconButton` instead, do
+`as={IconButton}`. This element merges all props provided by the
+[usePopupTargetButton](#usepopuptargetbutton) hook which would be used in place of this component if
+desired.
+
+```tsx
+const model = usePopupModel();
+
+// using this component
+<Popup.Target>Show Popup</PopupTarget>
+
+// using props instead
+const popupTargetButtonProps = usePopupTargetButton(model);
+<Button {...popupTargetButtonProps}>Show Popup</Button>
+```
+
+## Popup.Popper
+
+A `Popup.Popper` is a [Popper](#popper) component that is hooked up to the `PopupModel`
+automatically.
+
+## Popup Model
+
+The PopupModel contains all state and events needed for Popup behaviors and components. It composes
+a `DisclosureModel` for showing and hiding content.
+
+### Popup Model State
+
+<ArgsTable of={PopupStateComponent} />
+
+### Popup Model Events
+
+<ArgsTable of={PopupEventsComponent} />
+
+### Popup Model Config
+
+<ArgsTable of={PopupModelConfigComponent} />
+
+## Hooks
+
+### usePopupStack
+
+```ts
+const stackRef = usePopupStack(forwardRef?: React.RefObject<HTMLElement>): React.RefObject<HTMLDivElement>
+```
+
+**Note:** If you're using `Popper`, you do not need to use this hook directly.
+
+This hook will add the `stackRef` element to the Popup stack on mount and remove on unmount. If you
+use `Popper`, the popper `stackRef` is automatically added/removed from the Popup stack. The Popup
+stack is required for proper z-index values to ensure Popups are rendered correct. It is also
+required for global listeners like click outside or escape key closing a popup. Without the Popup
+stack, all popups will close rather than only the topmost one.
+
+If `forwardRef` is provided, it will be the same as `stackRef`. If `forwardRef` is not provided`,
+this hook will create one and return it.
+
+This hook should be used by all stacked UIs unless using the `Popper` component.
+
+Example:
+
+```tsx
+const model = usePopupModel();
+usePopupStack(model.state.stackRef, model.state.targetRef);
+
+// add some popup functionality
+useCloseOnOutsideClick(model);
+useCloseOnEscape(model);
+
+return (
+  <>
+    <button ref={model.state.targetRef}>Open Popup</button>
+    {model.state.visible
+      ? ReactDOM.createPortal(<div>Popup Contents</div>, model.state.stackRef.current)
+      : null}
+  </>
+);
+```
+
+### usePopup
+
+Convenience hook for common Popups used as non-modal dialogs. It provides props to mix into
+composite parts of the Popup pattern.
+
+> Note: This is **deprecated**. Please upgrade to using `usePopupModel` and the `Popup` compound
```

Comment:
I'm still confused about `usePopup` and `usePopupOld` if they're both deprecated haha. I still say let's just drop this. People can always use CK v5 with `@worday/canvas-kit-react-popup@4.x` if they need the old stuff temporarily.

Reply:
Hmm. That's fair. `usePopupOld` is a duplicate during testing.